pkg/cloud/api/mutable_resource.go

```diff
@@ -195,11 +195,41 @@ func (u *mutableResource[GA, Alpha, Beta]) CheckSchema() error {
 	if err != nil {
 		return err
 	}
+	ga, _ := u.ToGA()
+	alpha, _ := u.ToAlpha()
+	err = checkSubsetOf(ga, alpha)
```

Comment:
Don't you need to `fill` the struct first? otherwise this will ignore nil sub structs? Add test cases for that if they don't exist.

Reply:
in checkSubsetOf I leverage diff function to reduce code duplication but in fact what diff does is checking fields type and if it match then it comparing values. If I have unset field and they are not present in type B it will be cache in https://github.com/GoogleCloudPlatform/k8s-cloud-provider/pull/166/files/271e930ec4be6ca5aea04bf26a9a2077176d4195. Structs are treated as any other fields here. 
I added test case for struct.

Reply:
The link you gave doesn't work. What I'm asking about is this:

```
type A struct {
  B *B
}

type B struct {
  ...
  C *C
}

a := &A{}
```

if `a.B == nil`, does this code go and check the schema of the fields below, e.g. `a.B.C`

Reply:
you are right, this scenario was not working. I decided to create a new function based on reflect.Type there is no benefit from sharing the diff function in this case.